src/main/java/domain/Prizes.java

```diff
@@ -0,0 +1,50 @@
+package domain;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class Prizes {
+    private static final int INITIAL_PRIZE_COUNT = 0;
+
+    private final Map<PrizeCondition, Prize> prizes = new HashMap<PrizeCondition, Prize>() {{
+        put(PrizeCondition.FIRST, new Prize(PrizeCondition.FIRST, INITIAL_PRIZE_COUNT));
+        put(PrizeCondition.THIRD, new Prize(PrizeCondition.THIRD, INITIAL_PRIZE_COUNT));
+        put(PrizeCondition.FOURTH, new Prize(PrizeCondition.FOURTH, INITIAL_PRIZE_COUNT));
+        put(PrizeCondition.FIFTH, new Prize(PrizeCondition.FIFTH, INITIAL_PRIZE_COUNT));
+    }};
```

Comment:
List 를 사용하는 것에 대해서는 어떻게 생각하시나요? 😄  
lottos 를 상태값으로 가지고 있을 필요가 있을까요? 🤔 
생성자에서 prizes 의 값을 할당해주면 어떨까요? 😄 
(의견이 다른 부분은 편하게 말씀주세요! 😄 )
```suggestion
public class Prizes {

    private final List<Prize> prizes = new ArrayList<>();

    public Prizes(Lottos lottos, Lotto winningNumber) {
        // prizes 생성
    }
```

Reply:
lottos를 굳이 상태값으로 가지고 있을 필요는 없을 것 같아 profitRate에서 인자로 받아 처리하는 방식으로 변경했습니다 👍

생성자에서 값을 할당하는 경우, 저번 강의에서 생성자에선 최소한의 일만 하는 것이 좋다고 언급하셔서 생성자에서 제외를 시켰는데요. 이렇게 하든 저렇게 하든 Prizes instance가 생성될 때 만들어지는 거라 어떤 방법이 더 나을지는 저도 판단이 서질 않네요. 생성자에서 해도 무관할 거 같지만 큰 차이는 없을 것 같아 그대로 두었습니다.

List를 사용해도 괜찮지만 matchingNumber를 토대로 Prize instance를 찾게되는데, 개수는 얼마되지 않고 로또 당첨 순위도 증가될 가능성도 크지 않지만 O(1)로 곧바로 리턴하는게 더 좋을 것 같아 map으로 구현해보았습니다. list를 사용해서 filter로 걸어도 좋을 것 같습니다 👍